src/main/java/org/folio/rs/service/CheckInItemService.java

```diff
@@ -0,0 +1,47 @@
+package org.folio.rs.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.rs.client.CirculationClient;
+import org.folio.rs.client.LocationClient;
+import org.folio.rs.domain.dto.CheckInCirculationRequest;
+import org.folio.rs.domain.dto.CheckInItem;
+import org.folio.rs.repository.LocationMappingsRepository;
+import org.joda.time.DateTime;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+import static org.joda.time.DateTimeZone.UTC;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class CheckInItemService {
+
+  private final CirculationClient circulationClient;
+  private final LocationMappingsRepository locationMappingsRepository;
+  private final LocationClient locationClient;
+
+  public HttpStatus checkInItemByBarcode(String remoteStorageConfigurationId, CheckInItem checkInItem) {
```

Comment:
Why do we need HttpStatus here? If smth goes wrong inside this method Feign should throw exception.

Reply:
Could be return not successful status, but without any exceptions.

Reply:
In case of smth goes wrong feign client implementation throws exception and this validation will unuseful. You can see this if improve your unit tests just to cover this part of logic https://sonarcloud.io/component_measures?id=org.folio%3Amod-remote-storage&metric=new_coverage&pullRequest=37&selected=org.folio%3Amod-remote-storage%3Asrc%2Fmain%2Fjava%2Forg%2Ffolio%2Frs%2Fcontroller%2FCheckInRetrieveController.java&view=list

Reply:
Logic to check status was removed as feign client will take care about it.

Reply:
Logic to check status was removed as feign client will take care about it.